internal/provider/user_role_grant_resource_test.go

```diff
@@ -498,3 +499,221 @@ resource "cockroach_user_role_grant" "a_duplicate_grant" {
+	var (
+		orgGrantResourceName = "cockroach_user_role_grant.org_grant"
+	)
+
+	checkOrgRole := testRoleGrantExists(orgGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR), string(client.RESOURCETYPETYPE_ORGANIZATION))
+
+	var serviceAccountIDToDelete string
+
+	resource.Test(t, resource.TestCase{
+		IsUnitTest:               useMock,
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: getTestRoleGrantResourceWithNewUserConfig(),
+				Check:  resource.ComposeTestCheckFunc(checkOrgRole, getServiceAccountID(&serviceAccountIDToDelete)),
+			},
+			// In the acceptance test, this step simulates the external deletion
+			// of the service account created in the previous step. (In the
+			// integration test, we mock a request that returns a `not found`
+			// error instead.) Afterward, the state is refreshed, allowing
+			// Terraform to detect drift and update the plan accordingly.
+			{
+				PreConfig: func() {
+					if useMock {
+						return
+					}
+					deleteServiceAccountExternally(t, serviceAccountIDToDelete)
+				},
+				RefreshState:       true,
+				ExpectNonEmptyPlan: true,
+			},
+			// This step reapplies the exact same configuration. Because drift was
+			// detected in the previous step, Terraform will create a new service
+			// account.
+			{
+				Config: getTestRoleGrantResourceWithNewUserConfig(),
+				Check:  resource.ComposeTestCheckFunc(checkOrgRole),
+			},
+		},
+	})
+}
+
+func getTestRoleGrantResourceWithNewUserConfig() string {
+	return `
+resource "cockroach_service_account" "test_sa" {
+  name        = "Test cluster SA"
+  description = "A service account used for managing access to the test cluster"
+}
+
+resource "cockroach_user_role_grant" "org_grant" {
+	user_id = cockroach_service_account.test_sa.id
+	role = {
+		role_name = "BILLING_COORDINATOR",
+		resource_type = "ORGANIZATION",
+		resource_id = ""
+	}
+}
+`
+}
+
+func getServiceAccountID(serviceAccountID *string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		serviceAccountResourceName := "cockroach_service_account.test_sa"
+		resources := s.RootModule().Resources
+		resource, ok := resources[serviceAccountResourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", serviceAccountResourceName)
+		}
+
+		userID := resource.Primary.Attributes["id"]
+		*serviceAccountID = userID
+		return nil
+	}
+}
+
+func deleteServiceAccountExternally(t *testing.T, serviceAccountID string) {
+	if serviceAccountID == "" {
+		t.Fatalf("expected service account ID to be set")
+	}
+	p := testAccProvider.(*provider)
+	p.service = NewService(cl)
+	_, httpResp, err := p.service.DeleteServiceAccount(context.Background(), serviceAccountID)
+	if err != nil || httpResp.StatusCode != http.StatusOK {
+		t.Error(err != nil)
+		t.Error(httpOk.StatusCode)
+		t.Fatalf("failed to delete service account")
```

Comment:
It seems like these are a bit vague and different to how we commonly do error handling.  Can we use require here?  I guess similarly for the t.Fatalf above.
```suggestion
	require.NoError(t, err)
	require.Equal(t, httpResponse.StatusCode, http.StatusOK)
```

Reply:
Updated